tests/Foundation/FoundationApplicationTest.php

```diff
@@ -473,6 +484,44 @@ public function register()
     }
 }
 
+interface SampleInterface
+{
+    public function getPrimitive();
+}
+
+class SampleImplementation implements SampleInterface
+{
+    private $primitive;
+
+    public function __construct($primitive)
+    {
+        $this->primitive = $primitive;
+    }
+
+    public function getPrimitive()
+    {
+        return $this->primitive;
+    }
+}
+
+class InterfaceToImplementationDeferredServiceProvider extends ServiceProvider implements DeferrableProvider
+{
+    public function register()
+    {
+        $this->app->bind(SampleInterface::class, SampleImplementation::class);
```

Comment:
This simple can't work. You need to bind the primitive against the container otherwise it can't be resolved. I have no idea how this could ever have worked.